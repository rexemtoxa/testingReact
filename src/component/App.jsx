import React from 'react';
import TabsConstructor from './TabsConstructor';

const App = () => <TabsConstructor />;

export default App;
