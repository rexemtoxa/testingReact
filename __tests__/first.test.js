import React from 'react';
import { mount, shallow } from 'enzyme';
import App from '../src/component/App';

test('snapshot testing to toggle tab', () => {
  const wrapper = mount(<App />);
  expect(wrapper.render()).toMatchSnapshot();
  const tabs = wrapper.find('[data-test="tab-anchor"]');
  const tab2 = tabs.at(2);
  tab2.simulate('click');
  expect(wrapper.render()).toMatchSnapshot();
});

test('jest-enzyme testyng to toggle tab', () => {
  const wrapper = mount(<App />);
  const tabs = wrapper.find('li[data-test="tab-anchor"]');
  const firstTab = tabs.at(0);
  const secondTab = tabs.at(1);

  expect(firstTab).toHaveClassName('react-tabs__tab react-tabs__tab--selected');
  expect(secondTab).not.toHaveClassName('react-tabs__tab--selected');
});

test('jest-enzyme testyng to toggle tab 2', () => {
  const wrapper = mount(<App />);
  wrapper.find('li[data-test="tab-anchor"]').at(1).simulate('click');

  const tabs = wrapper.find('li[data-test="tab-anchor"]');
  const firstTab = tabs.at(0);
  const secondTab = tabs.at(1);
  expect(secondTab).toHaveClassName('react-tabs__tab react-tabs__tab--selected');
  expect(firstTab).not.toHaveClassName('react-tabs__tab--selected');
});
