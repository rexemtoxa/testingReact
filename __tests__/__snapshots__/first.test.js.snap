// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing to toggle tab 1`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="true"
      class="react-tabs__tab react-tabs__tab--selected"
      data-test="tab-anchor"
      id="react-tabs-0"
      role="tab"
      tabindex="0"
    >
       1 
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      data-test="tab-anchor"
      id="react-tabs-2"
      role="tab"
    >
       2 
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-0"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-1"
    role="tabpanel"
  >
    <h2>
      Any content 1
    </h2>
  </div>
  <div
    aria-labelledby="react-tabs-2"
    class="react-tabs__tab-panel"
    id="react-tabs-3"
    role="tabpanel"
  />
</div>
`;

exports[`snapshot testing to toggle tab 2`] = `
<div
  class="react-tabs"
  data-tabs="true"
>
  <ul
    class="react-tabs__tab-list"
    role="tablist"
  >
    <li
      aria-controls="react-tabs-1"
      aria-disabled="false"
      aria-selected="false"
      class="react-tabs__tab"
      data-test="tab-anchor"
      id="react-tabs-0"
      role="tab"
    >
       1 
    </li>
    <li
      aria-controls="react-tabs-3"
      aria-disabled="false"
      aria-selected="true"
      class="react-tabs__tab react-tabs__tab--selected"
      data-test="tab-anchor"
      id="react-tabs-2"
      role="tab"
      tabindex="0"
    >
       2 
    </li>
  </ul>
  <div
    aria-labelledby="react-tabs-0"
    class="react-tabs__tab-panel"
    id="react-tabs-1"
    role="tabpanel"
  />
  <div
    aria-labelledby="react-tabs-2"
    class="react-tabs__tab-panel react-tabs__tab-panel--selected"
    id="react-tabs-3"
    role="tabpanel"
  >
    <h2>
      Any content 2
    </h2>
  </div>
</div>
`;
